src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitCyclesCheckTest.java

```diff
@@ -0,0 +1,47 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2022 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.internal;
+
+import static com.tngtech.archunit.library.dependencies.SlicesRuleDefinition.slices;
+
+import org.junit.jupiter.api.Test;
+
+import com.tngtech.archunit.core.domain.JavaClasses;
+import com.tngtech.archunit.core.importer.ClassFileImporter;
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.lang.ArchRule;
+import com.tngtech.archunit.library.freeze.FreezingArchRule;
+
+public class ArchUnitCyclesCheckTest {
+
+    @Test
+    public void testSlicesShouldBeFreeOfCycle() {
```

Comment:
Please do this: imagine you are a new contributor, and execution of this test has generated a new suppression. Then, in the issue, place links and information about the concept of this test and links to freeze documentation. After that, place a Javadoc here that explains how this test works, and a link to your comment in the issue.

Reply:
Yes, it good idea to read details in actual test class. Hints on how to interpretate is also good.

What is the process if we ok to allow extra cycle?

Reply:
@nrmancuso Please take a look at the javadoc.

Reply:
does it mean that test will always pass and just update freeze file, so will enforce no cycles by simple PR review and noticing a diff - I am ok.

If user forget to place freeze file to commit, CI will fail as it will detect  a git-diff.

Reply:
Yes, for any new violation, the store will be automatically updated, if the user commits the changes in the store, they will be seen and reviewed in the PR.

> If user forget to place freeze file to commit, CI will fail as it will detect a git-diff.

Yes

Reply:
please mention that change in freeze file need to be placed in commit.
Any change in freeze file should be committed and shared in PR.

Reply:
> Whenever a new violation is introduced, the test will pass successfully but the frozen violations will be updated. In that case, developers should check if the new violation is expected or not. In both cases create a separate commit in the PR for discussing changes in the violation store.

Added this para to the javadoc.